.github/workflows/build.yml

```diff
@@ -5,39 +5,84 @@ on: [push, pull_request]
 jobs:
   build:
     runs-on: ${{ matrix.host_os }}
-    container: ${{ matrix.container }}
     strategy:
       matrix:
         target_os: [linux]
         host_os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
-        container: ['']
         include:
           - target_os: windows
-            host_os: ubuntu-latest
-            container: krzysh/colobot-build:latest
+            host_os: windows-2019
       fail-fast: false
     steps:
-    - name: Install Colobot dependencies
-      run: sudo apt-get update && sudo apt-get install -y --no-install-recommends build-essential cmake libsdl2-dev libsdl2-image-dev libsdl2-ttf-dev libsndfile1-dev libvorbis-dev libogg-dev libpng-dev libglew-dev libopenal-dev libboost-dev libboost-system-dev libboost-filesystem-dev libboost-regex-dev libphysfs-dev gettext git po4a vorbis-tools librsvg2-bin xmlstarlet
-      if: matrix.container == ''
     - uses: actions/checkout@v2
+    - name: Checkout the Google Test submodule
+      run: git submodule update --init -- lib/googletest
+    - name: Install Colobot dependencies (for Linux)
+      run: sudo apt-get update && sudo apt-get install -y --no-install-recommends build-essential cmake libsdl2-dev libsdl2-image-dev libsdl2-ttf-dev libsndfile1-dev libvorbis-dev libogg-dev libpng-dev libglew-dev libopenal-dev libboost-dev libboost-system-dev libboost-filesystem-dev libboost-regex-dev libphysfs-dev gettext git po4a vorbis-tools librsvg2-bin xmlstarlet
+      if: matrix.target_os == 'linux'
+    - name: Install Ninja
+      uses: seanmiddleditch/gha-setup-ninja@master
+      if: matrix.target_os == 'windows'
```

Comment:
I no longer like how the `if` steps work. I was hoping we could eventually replace them with platform-independent actions like `run-cmake`, but it seems like this is going in exactly the opposite direction. Perhaps it would be cleaner to separate this into `build_linux` and `build_windows` instead, or somehow extract the platform-dependent dependency setup elsewhere?

Reply:
I agree. Was thinking about splitting the builds up but was too busy trying to make it work in the first place.